src/exporter.js

```diff
@@ -67,80 +77,64 @@ function Exporter (hash, dagService, options) {
         })
         series(array, (err, res) => {
           if (err) {
-            callback()
+            rs.emit('error', err)
             return
           }
           rs.push(null)
-          callback()
           return
         })
       }
       this.push({ content: rs, path: name })
-      callback()
-      return
+      done()
     }
   }
 
-  let dirExporter = (node, name, callback) => {
-    let init
+  // Logic to export a unixfs directory.
+  let dirExporter = (node, name, add, done) => {
+    if (!add) throw new Error('add must be set')
```

Comment:
```add``` will always be set since it's a param called from ```visit()```  which is afaik not meant to be exposed